### ISSUE TITLE
fits tables with format identifier "X" do not read in properly

### ISSUE THREAD

**Issue by hamogu** [atpy/atpy#12]
I try to read a fits file (a Chandra event list actually), which contains one column of status bits.
When I read that I get 
    /usr/local/lib/python2.6/dist-packages/ATpy-0.9.4-py2.6.egg/atpy/fitstable.pyc in read(self, filename, hdu, verbose)

```
160         self.data[name][:] = data[:]

ValueError: shape mismatch: objects cannot be broadcast to a single shape
```

I have traced that down to an incorrect conversion of the format identifier.
    (Pdb) hdu.columns[i].format, hdu.columns[i].bzero
    ('32X', None)
    (Pdb) type.type
    <type 'numpy.uint8'>
    (Pdb) shape
    (4,)
    (Pdb) data.dtype
    dtype('bool')
    (Pdb) data.shape
    (2364114, 32)
    (Pdb) self.data[name].shape
    (2364114, 4)
    (Pdb) hdu.columns[i].format
    '32X'
    (Pdb) format
    'X'
    (Pdb) self.data[name].dtype
    dtype('uint8')

**Comment by astrofrog** [comment]
I actually ran into the exact same issue recently with a Chandra Event list and fixed this in 1facd869b7a30ae8fd3c72232ea8e8277118761f a couple of weeks ago - could you try installing the latest source version of ATpy and see if you are still running into this issue?

**Comment by hamogu** [comment]
The problem is solved with the current version. Great.
(In practice I do a filtering on that file anyway before I read it with atpy, thus reducing the table I read into python from 78M to 140k, so it is not a severe issue for me anyway.)
